skxray/fitting/xrf_model.py

```diff
@@ -0,0 +1,1042 @@
+        else:
+            logger.critical(' No element is selected for fitting!')
+
+        self.incident_energy = self.parameter['coherent_sct_energy']['value']
+        self.set_compton_model()
+        self.set_elastic_model()
+
+    def set_compton_model(self):
+        """
+        setup parameters related to Compton model
+        """
+        compton = ComptonModel()
+
+        compton_list = ['coherent_sct_energy', 'compton_amplitude',
+                        'compton_angle', 'fwhm_offset', 'fwhm_fanoprime',
+                        'e_offset', 'e_linear', 'e_quadratic',
+                        'compton_gamma', 'compton_f_tail',
+                        'compton_f_step', 'compton_fwhm_corr',
+                        'compton_hi_gamma', 'compton_hi_f_tail']
+
+        logger.debug(' ###### Started setting up parameters for compton model. ######')
+        for name in compton_list:
+            if name in self.parameter.keys():
+                _set_parameter_hint(name, self.parameter[name], compton)
+            else:
+                _set_parameter_hint(name, self.parameter_default[name], compton)
+        logger.debug(' Finished setting up parameters for compton model.')
+
+        self.compton_param = compton.make_params()
+        self.compton = compton
+
+    def set_elastic_model(self):
+        """
+        setup parameters related to Elastic model
+        """
+        elastic = ElasticModel(prefix='elastic_')
+
+        item = 'coherent_sct_amplitude'
+        if item in self.parameter.keys():
+            _set_parameter_hint(item, self.parameter[item], elastic)
+        else:
+            _set_parameter_hint(item, self.parameter_default[item], elastic)
+
+        logger.debug(' ###### Started setting up parameters for elastic model. ######')
+
+        # set constraints for the following global parameters
+        elastic.set_param_hint('e_offset', value=self.compton_param['e_offset'].value,
+                               expr='e_offset')
+        #elastic.set_param_hint('e_offset', expr='e_offset')
+        elastic.set_param_hint('e_linear', value=self.compton_param['e_linear'].value,
+                               expr='e_linear')
+        elastic.set_param_hint('e_quadratic', value=self.compton_param['e_quadratic'].value,
+                               expr='e_quadratic')
+        elastic.set_param_hint('fwhm_offset', value=self.compton_param['fwhm_offset'].value,
+                               expr='fwhm_offset')
+        elastic.set_param_hint('fwhm_fanoprime', value=self.compton_param['fwhm_fanoprime'].value,
+                               expr='fwhm_fanoprime')
+        elastic.set_param_hint('coherent_sct_energy', value=self.compton_param['coherent_sct_energy'].value,
+                               expr='coherent_sct_energy')
+        logger.debug(' Finished setting up parameters for elastic model.')
+        self.elastic = elastic
+
+    def set_element_model(self, ename, default_area=1e5):
```

Comment:
This is just a default number I use to set the initial intensity, according to some experience. We need to give a starting value for fitting.